impl/src/main/java/com/sun/faces/util/Util.java

```diff
@@ -760,44 +797,17 @@ public static StateManager getStateManager(FacesContext context) throws FacesExc
         return context.getApplication().getStateManager();
     }
 
-    public static Class getTypeFromString(String type) throws ClassNotFoundException {
-        Class result;
-        switch (type) {
-        case "byte":
-            result = Byte.TYPE;
-            break;
-        case "short":
-            result = Short.TYPE;
-            break;
-        case "int":
-            result = Integer.TYPE;
-            break;
-        case "long":
-            result = Long.TYPE;
-            break;
-        case "float":
-            result = Float.TYPE;
-            break;
-        case "double":
-            result = Double.TYPE;
-            break;
-        case "boolean":
-            result = Boolean.TYPE;
-            break;
-        case "char":
-            result = Character.TYPE;
-            break;
-        case "void":
-            result = Void.TYPE;
-            break;
-        default:
-            if (type.indexOf('.') == -1) {
-                type = "java.lang." + type;
-            }
-            result = Util.loadClass(type, Void.TYPE);
-            break;
+    public static Class<?> getTypeFromString(String type) throws ClassNotFoundException {
+        if ( type == null ) throw new ClassNotFoundException("Type is required");
```

Comment:
This changes the previous behavior.

Reply:
switch over a null value raise a NPE before Java 21

so the previous result was `NullPointerException` which is imho less explicative...

Reply:
Just change the message. You can use the utility method Objects.requireNonNull() for this.

Reply:
ok